exporter/datadogexporter/factory_test.go

```diff
@@ -295,19 +296,18 @@ func TestOnlyMetadata(t *testing.T) {
 
 	factory := NewFactory()
 	ctx := context.Background()
-	cfg := &Config{
+	cfg := &datadogconfig.Config{
 		ClientConfig:  defaultClientConfig(),
 		BackOffConfig: configretry.NewDefaultBackOffConfig(),
 		QueueSettings: exporterhelper.NewDefaultQueueConfig(),
 
-		API:          APIConfig{Key: "aaaaaaa"},
-		Metrics:      MetricsConfig{TCPAddrConfig: confignet.TCPAddrConfig{Endpoint: server.URL}},
-		Traces:       TracesConfig{TCPAddrConfig: confignet.TCPAddrConfig{Endpoint: server.URL}},
+		API:          datadogconfig.APIConfig{Key: "aaaaaaa"},
+		Metrics:      datadogconfig.MetricsConfig{TCPAddrConfig: confignet.TCPAddrConfig{Endpoint: server.URL}},
+		Traces:       datadogconfig.TracesExporterConfig{TCPAddrConfig: confignet.TCPAddrConfig{Endpoint: server.URL}},
 		OnlyMetadata: true,
 
-		HostMetadata: HostMetadataConfig{
+		HostMetadata: datadogconfig.HostMetadataConfig{
 			Enabled:        true,
-			HostnameSource: HostnameSourceFirstResource,
```

Comment:
Is there a reason why we're removing HostnameSource here?

Reply:
This is deprecated in https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/39069, I should have removed it in the previous PR. It's leading to deprecation linter failures now